CIP-XXXX/README.md

```diff
@@ -0,0 +1,364 @@
+of the project and therefore any potential metadata associated with said CIPs ignored from the time of the registration
+forward.
+
+**Example:**
+
+`[25, 27]`
+
+#### 3. Witness Method
+
+The witness method portion of the payload should describe the method used to generate the witness signature (CIP-8, etc)
+to aid in validation by third parties.
+
+**Example:**
+
+`['CIP-8','Ed25519']`
+
+#### 4. Nonce
+
+As with CIP-36 Voter Registration method, we must include a piece of data that changes to avoid a replay attack. As with
+CIP-36, and stake pool KES counter rotation, the nonce is expected to be an integer value that is incremented as new or
+updated registrations occur. The suggestion to use the current slot tip of the network at the time of submission from 
+CIP-36 is also recommended here.
+
+Validators and indexers observing token registrations should always honor the verified registration with the highest
+nonce value.
+
+**Example:**
+
+`12345`
+
+### Optional Fields
+
+The following are optional fields that may be included in the token registration to provide additional information or
+context to the registration.
+
+#### 5. Data Oracle URI
+
+To be utilized and expanded upon in a separate CIP, this should be a valid URI pointing to a source of additional,
+potentially dynamic information relating to the project and/or the tokens minted under it.
+
+**Example:**
+
+`'https://oracle.mytokenproject.io/'`
+
+#### 6. CIP-Specific Information
+
+This entry, if present, should be a CIP ID indexed object containing additional information pertaining to that CIP.
+These additional CIPs/Schemas to be determined by the community could include:
+
+- CIP-25/68 NFT Project top-level project information
+- CIP-26 FT Project monetary policy information
+
+**Example:**
+
+```json 
+{
+  "25": {
+    "project_name": "Cool NFT Project",
+    "description": [
+      "This is a description of my project", 
+      "longer than 64 characters so broken up into a string array"
+    ],
+    "banner_image_uri": "https://static.coolnftproject.io/images/banner1.jpg",
+    "icon_image_uri": "https://static.coolnftproject.io/images/icon.png",
+    "social_media": {
+      "discord": "https://coolnftproject.discord.gg/",
+      "twitter": "https://twitter.com/coolNFTproject"
+    }
+  },
+  "27": {
+    "rate": "0.05",
+    "addr": [
+      "addr_test1qqp7uedmne8vjzue66hknx87jspg56qhkm4gp6ahyw7kaahevmtcux",
+      "lpy25nqhaljc70094vfu8q4knqyv6668cvwhsq64gt89"
+    ]
+  }
+}
+```
+
+### Token Registration Witness Array
+
+The Witness Array included in the on-chain metadata should consist of an array of arrays with two elements, the public
+key hash of the signing key (as included in the native script) and the signed key witness. If a script requires multiple
+signatures, enough signatures to meet the criteria of the script should be included and required for proper validation
```

Comment:
Can this lead to ambiguities when for a 2 out of [A, B, C] script A+B and B+C both individually push a different update with the same nonce?

Reply:
Hrm, that is a potential edge-case scenario that would need to be tested and developed against. Would potentially have to resort to accepting the "first seen" in that case?

Reply:
I guess it would not happen "by chance", more like in a situation with B as kind of a Byzantine assassin, so B would trick A and C individually to sign different multi-sig transactions and then submit them within the same block. Maybe enough to take block height + tx hash?

Reply:
If the nonce will be a hash then the id of any of the input utxos is guaranteed to be unique

Reply:
As described, the nonce should be a "counter value" [uint] so you could always reference the most recent update (largest integer) and avoid a historic "replay" attack.

Reply:
@Crypto2099 @nullhashpixel looks like you have addressed this question, so resolving.